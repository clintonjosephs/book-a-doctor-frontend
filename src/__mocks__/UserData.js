const UserData = {
  name: 'Test User',
  email: 'admin@microverse.com',
  role: 'admin',
  image_url: 'http://localhost:3001/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--d3e44c619bcee024384c34673e88be7a2547c5f8/194934.png',
  created_at: '2022-06-03T11:13:52.186Z',
  updated_at: '2022-06-03T11:13:52.619Z',
};

export default UserData;
