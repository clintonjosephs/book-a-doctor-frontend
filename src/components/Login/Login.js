import React from 'react';
import styles from './Login.module.css';

const Login = () => {
  const loginHandler = () => {
    console.log('someone hit submit button');
  };

  return (
    <section className={styles.loginContainer}>
      <div className={`${styles.login} ${styles.shadow}`}>
        <h2>Login</h2>
        <div className={styles.line} />
        <form onSubmit={loginHandler}>
          <input
            type="text"
            name="email"
            placeholder="Email Address"
            required
          />
          <input
            type="password"
            name="password"
            placeholder="Password"
            required
          />
          <div>
            <input
              type="checkbox"
              name="remember"
            />
            {' '}
            &nbsp;
            Remember me?
          </div>
          <button type="submit" className={styles.formSubmitBtn}>Log in</button>
        </form>
      </div>
    </section>
  );
};

export default Login;
