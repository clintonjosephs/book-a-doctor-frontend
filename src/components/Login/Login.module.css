.loginContainer {
  width: 100%;
  height: 100vh;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #fff;
}

.login {
  width: 50rem;
  padding: 1rem;
  border-radius: 0.5rem;
}

.line {
  width: 7rem;
  height: 0.2rem;
  background-color: var(--primary-color);
  margin-top: 0.5rem;
  margin-bottom: 2rem;
}

.shadow {
  -moz-box-shadow: 0 0 3px #ccc !important;
  -webkit-box-shadow: 0 0 3px #ccc !important;
  box-shadow: 0 0 35px #ccc !important;
}

.formSubmitBtn {
  width: 7rem;
  height: 3rem;
  border-radius: 0.5rem;
  border: none;
  background: var(--accent-color);
  font-size: 1rem;
  color: var(--white);
}

.formSubmitBtn:hover {
  background: var(--primary-color);
  color: var(--white);
}

form {
  display: flex;
  flex-direction: column;
}

input[type="text"],
input[type="password"] {
  width: 100%;
  height: 3rem;
  padding: 0.5rem;
  margin-bottom: 1rem;
  border: solid 1px var(--accent-color);
  font-size: 1rem;
}

form > div {
  margin-bottom: 1rem;
}

input[type="text"]:focus,
input[type="password"]:focus {
  outline: none;
  box-shadow: none;
  border: solid 1px var(--primary-color) !important;
}

.banner {
  width: 100%;
  min-height: 20rem;
  margin-bottom: 2rem;
}

.success {
  color: green;
  text-align: center;
  margin-bottom: 0.5rem;
}

.failure {
  margin-bottom: 0.5rem;
  text-align: center;
  color: red;
}

@media (max-width: 768px) {
  .login {
    width: 100%;
  }

  .banner {
    width: 100%;
    margin-bottom: 0;
  }
}
