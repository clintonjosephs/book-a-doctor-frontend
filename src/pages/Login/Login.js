import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { accountLogin } from '../../redux/bookDoctor/doctorThunks';
import styles from './Login.module.css';
import banner from '../../assets/images/banner.jpeg';

const Login = () => {
  const { loading } = useSelector((state) => state.bookDoctorReducer);
  const [info, setMessage] = useState({ message: '', status: false });

  const setup = () => ({
    email: '',
    password: '',
  });

  const dispatch = useDispatch();

  const [loginData, setLoginData] = useState(setup());

  const { email, password } = loginData;

  const changeHandler = (event) => {
    setLoginData((prevState) => ({
      ...prevState,
      [event.target.name]: event.target.value,
    }));
  };

  const loginHandler = async (event) => {
    event.preventDefault();
    const response = await dispatch(accountLogin(loginData));
    setMessage(response);
  };

  const { message, status } = info;

  return (
    <>
      <section className={styles.loginContainer}>
        <img src={banner} alt="login banner" className={styles.banner} />
        <div className={`${styles.login} ${styles.shadow}`}>
          <h2>Book a Doctor</h2>
          <div className={styles.line} />
          <div className={`${status === true ? styles.success : styles.failure}`}>{message}</div>
          <form onSubmit={loginHandler}>
            <input
              type="text"
              name="email"
              placeholder="Email Address"
              required
              value={email}
              onChange={changeHandler}
            />
            <input
              type="password"
              name="password"
              placeholder="Password"
              value={password}
              required
              onChange={changeHandler}
            />
            <div>
              <input
                type="checkbox"
                name="remember"
              />
              {' '}
              &nbsp;
              Remember me?
            </div>
            {
              loading ? <button type="submit" className={styles.formSubmitBtn}>Log in</button> : <button type="button" className={styles.formSubmitBtn}>Processing ...</button>
            }

            <div style={{ marginTop: '1rem' }}>
              Don&apos;t have an account?
              <NavLink to="/signup"> signup </NavLink>
            </div>
          </form>
        </div>
      </section>
    </>
  );
};

export default Login;
