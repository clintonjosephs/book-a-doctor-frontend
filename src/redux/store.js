import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
