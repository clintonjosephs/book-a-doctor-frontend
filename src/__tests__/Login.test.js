import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import store from '../redux/store';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from '../pages/Login/Login';
import bookDoctorReducer, * as actions from '../redux/bookDoctor/doctorActions';
import UserData from '../__mocks__/UserData';

const LoginProvider = () => (
  <Provider store={store}>
    <Router>
      <Login />
    </Router>
  </Provider>
);


afterEach(() => {
    cleanup();
});


describe('check login components', () => {
  it('matches actual design', () => {
    const component = renderer.create(<LoginProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Test Login asynchronous actions', () => {
  // initial state
  const initalState = {
    doctors: [],
    appointments: [],
    userDetails: {},
    error: false,
    loading: true,
  };

  const dataPush = bookDoctorReducer(
    initalState,
    actions.userDetails(UserData)
  );

  it('has user details', () => {
    expect(dataPush.userDetails).toEqual(UserData);
  });

  it('has the correct user', () => {
    const { name, email, role } = dataPush.userDetails;
    expect(name).toEqual('Test User');
    expect(email).toEqual('admin@microverse.com');
    expect(role).toEqual('admin');
  });
});

