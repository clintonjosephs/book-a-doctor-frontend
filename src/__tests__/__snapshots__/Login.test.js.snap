// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check login components matches actual design 1`] = `
<div
  style={
    Object {
      "opacity": 1,
      "transform": "translateX(300px) translateZ(0)",
    }
  }
>
  <section
    className="loginContainer"
  >
    <div
      className="login shadow"
    >
      <h2>
        Book a Doctor
      </h2>
      <div
        className="line"
      />
      <div
        className="failure"
      >
        
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          data-testid="email-input"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="email"
          value=""
        />
        <input
          data-testid="password-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <div>
          <input
            name="remember"
            type="checkbox"
          />
           
          Â  Remember me?
        </div>
        <button
          className="formSubmitBtn"
          type="submit"
        >
          Log in
        </button>
        <div
          style={
            Object {
              "marginTop": "1rem",
            }
          }
        >
          Don't have an account?
          <a
            className=""
            href="/signup"
            onClick={[Function]}
          >
             signup 
          </a>
        </div>
      </form>
    </div>
  </section>
</div>
`;
